### PR TITLE
Fix for visibility of close button

### PR DESCRIPTION
**Pull request by juhl**
Fixed some bugs.

- Close button was invisible just after calling addToolWindow even when AlwaysDisplayFullTabs is on.
- Example project have a bug in checkable actions in menu.

### CODE REVIEW

**Comment by baldurk** [comment]
Hi, thanks for this.

I understand the example project bugfix - I don't use it myself, I only separated this lib from [renderdoc](https://github.com/baldurk/renderdoc) for convenience, so I always use that as the test case.

I'm not sure I follow the fix for the close button though. It seems like your change will only work if you pass in the `ToolWindowManager::AlwaysDisplayFullTabs` property with the new `addToolWindow` parameter, so that it's set before anything could possibly check the properties. Wouldn't the bug still exist if the user called `setToolWindowProperties` like before?

It seems like maybe what needs to happen is when the tool window properties are changed, we bubble up a refresh to anything that might want to change behaviour based on the properties. That way it will work regardless of what the user does.

**Comment by juhl** [comment]
You're right.
I've changed to setWindowProperties() reflect visibility of close button immediately.

File: example/MainWindow.cpp [hunk 1]
```diff
@@ -61,6 +61,14 @@ MainWindow::MainWindow(QWidget *parent) :
...
                                        ui->toolWindowManager->areaOf(toolWindows[4])));
 
+#if 0
+  ui->toolWindowManager->setToolWindowProperties(toolWindows[0], ToolWindowManager::AlwaysDisplayFullTabs);
+  ui->toolWindowManager->setToolWindowProperties(toolWindows[1], ToolWindowManager::AlwaysDisplayFullTabs);
+  ui->toolWindowManager->setToolWindowProperties(toolWindows[2], ToolWindowManager::AlwaysDisplayFullTabs);
+  ui->toolWindowManager->setToolWindowProperties(toolWindows[3], ToolWindowManager::AlwaysDisplayFullTabs);
+  ui->toolWindowManager->setToolWindowProperties(toolWindows[4], ToolWindowManager::AlwaysDisplayFullTabs);
+  ui->toolWindowManager->setToolWindowProperties(toolWindows[5], ToolWindowManager::AlwaysDisplayFullTabs);
+#endif
```
Comment (baldurk) [inline example/MainWindow.cpp]: Can you remove this (I guess you were using it for testing?). I think the example shouldn't have any stale/outdated code in it.

File: src/ToolWindowManager.cpp [hunk 2]
```diff
@@ -151,8 +151,8 @@ void ToolWindowManager::addToolWindows(QList<QWidget *> toolWindows, const ToolW
     toolWindow->hide();
     toolWindow->setParent(0);
     m_toolWindows << toolWindow;
-    m_toolWindowProperties[toolWindow] = ToolWindowProperty(0);
-    QObject::connect(toolWindow, &QWidget::windowTitleChanged, this, &ToolWindowManager::windowTitleChanged);
+    m_toolWindowProperties[toolWindow] = properties;
+    QObject::connect(toolWindow, &QWidget::windowTitleChanged, this, &ToolWindowManager::windowTitleChanged);
```
Comment (baldurk) [inline src/ToolWindowManager.cpp]: There's some added trailing whitespace here.

Comment (juhl) [reply src/ToolWindowManager.cpp]: Removed it in next commit.

File: example/MainWindow.cpp (hunk 1 above)
Comment (juhl) [reply example/MainWindow.cpp]: Removed it in next commit.

**Pull request merged**